haskell-src/lib/ChainwebData/Env.hs

```diff
@@ -224,12 +234,47 @@ data FillArgs = FillArgs
   { _fillArgs_delayMicros :: Maybe Int
   } deriving (Eq, Ord, Show)
 
-data ServerEnv = ServerEnv
-  { _serverEnv_port :: Int
-  , _serverEnv_runFill :: Bool
-  , _serverEnv_fillDelay :: Maybe Int
-  , _serverEnv_serveSwaggerUi :: Bool
-  } deriving (Eq,Ord,Show)
+data WorkerEnv = Full FullEnv | HTTP HTTPEnv | ETL ETLEnv
+  deriving (Eq,Ord,Show)
+
+type FullEnv = (HTTPEnv, ETLEnv)
+
+_Full :: Prism' WorkerEnv FullEnv
+_Full = prism' Full $ \case
+  Full x -> Just x
+  _ -> Nothing
+
+_HTTP :: Prism' WorkerEnv HTTPEnv
+_HTTP = prism' HTTP $ \case
+  HTTP x -> Just x
+  _ -> Nothing
+
+_ETL :: Prism' WorkerEnv ETLEnv
+_ETL = prism' ETL $ \case
+  ETL x -> Just x
+  _ -> Nothing
```

Comment:
What's the benefit of defining these accessors as `Prism'`s? As far as I can tell, we're only using them to read these values, so they could just be functions of type `WorkerEnv -> Maybe XXX`.

Actually, since we don't need `Prism'`s, we can make these accessors even more useful by replacing them with:
``` suggestion
getHTTPEnv :: WorkerEnv -> Maybe HTTPEnv
getHTTPEnv = \case
  Full (env,_) -> Just env
  HTTP env -> Just env
  ETL _ -> Nothing

getETLEnv :: WorkerEnv -> Maybe ETLEnv
getETLEnv = \case
  Full (_,env) -> Just env
  ETL env -> Just env
  HTTP _ -> Nothing
```